clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -1932,8 +1950,9 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::advance_region_of_interest(
         if (input.cur_region > 0) {
             VPRINT(this, 3, "skip_instructions input=%d: inserting separator marker\n",
                    input.index);
-            input.queue.push_back(record);
+            input.queue.emplace_back(record, is_record_real);
```

Comment:
Doesn't this need to update the count if is_record_real?  That helper is looking better now.